rollup.config.js

```diff
@@ -0,0 +1,252 @@
+ */
+function getOutput(file, type, name, exports, globals) {
+    let config = {
+        exports: exports,
+        name: name,
+        file: file,
+        sourcemap: true,
+        format: type
+    };
+
+    if (type === 'umd') {
+        config.globals = globals;
+    }
+
+    return config;
+}
+
+/**
+ * Returns the correct plugins array
+ *
+ * @method getPlugins
+ *
+ * @param {string} type
+ * @param {object} babelOptions
+ * @param {object} resolverOptions
+ *
+ * @returns {Array}
+ */
+function getPlugins(type, babelOptions, resolverOptions) {
+    const babelPlugin = babel(
+        getBabelConfig(
+            babelOptions.exclude,
+            babelOptions.targets,
+            babelOptions.forceAllTransforms,
+            babelOptions.transformRuntimeOptions,
+            babelOptions.minified
+        )
+    );
+
+    switch (type) {
+        case 'umd':
+            return [
+                resolve(
+                    Object.assign(
+                        {
+                            browser: true,
+                            preferBuiltins: true
+                        },
+                        resolverOptions
+                    )
+                ),
+                commonjs(),
+                babelPlugin,
+                nodeGlobals(),
+                json(),
+                builtins(),
+                terser({sourcemap: true}),
+                bundleSize()
+            ];
+        case 'cjs':
+        case 'es':
+            return [
+                babelPlugin,
+                json(),
+                autoExternal(),
+                cleanup()
+            ];
+    }
+}
+
+
+/**
+ * Returns the correct babel config
+ *
+ * @method getBabelConfig
+ *
+ * @param {Array|string} exclude
+ * @param {object} targets
+ * @param {boolean} forceAllTransforms
+ * @param {object} transformRuntimeOptions
+ * @param {boolean} minified
+ *
+ * @returns {{minified: boolean, presets: *[][], babelrc: boolean, plugins: *[][], runtimeHelpers: boolean, exclude: *}}
+ */
+function getBabelConfig(exclude, targets, forceAllTransforms = false, transformRuntimeOptions = {}, minified = false) {
+    let plugins = [
+        '@babel/plugin-proposal-export-default-from',
+        '@babel/plugin-proposal-export-namespace-from',
+        [
+            '@babel/plugin-transform-runtime',
+            Object.assign({useESModules: true}, transformRuntimeOptions)
+        ]
+    ];
+
+    if (process.env.ISTANBUL === 'true') {
+        plugins.push('istanbul');
+    }
+
+    return {
+        exclude: exclude,
+        babelrc: false,
+        runtimeHelpers: true,
+        minified: minified,
+        presets: [
+            [
+                '@babel/preset-env',
+                {
+                    forceAllTransforms: forceAllTransforms,
+                    modules: false,
+                    targets: targets
+                }
+            ]
+        ],
+        plugins: plugins
+    };
+}
```

Comment:
I really think this should be made more explicit. It's very hard to have the whole picture of what's going on, and this is one of the most error-prone things in the entire project.

Reply:
What should be made more explicit?

Reply:
I'm resolving this. It's duplicated.